packages/graphql_flutter/lib/src/widgets/hooks/query.dart

```diff
@@ -31,6 +31,9 @@ QueryHookResult<TParsed> useQuery<TParsed>(QueryOptions<TParsed> options) {
     query.stream,
     initialData: query.latestResult,
   );
+
+  query.onData(QueryCallbackHandler(options: options).callbacks);
```

Comment:
This'll add new callback handlers on every build, which is too many. But since callbacks are also removed after their invocation, adding those only on the options change might be too little.

I think we should do two things here. 

First, we can put this in an `useEffect`:

```suggestion
  useEffect(() {
    final cleanup = query.onData(
      QueryCallbackHandler(options: options).callbacks,
      removeAfterInvocation: false,
    );  
    return cleanup;
  }, [options, query]);
```
This way we'll register new callbacks when the options (or query) instances change and we'll handle removing them as well.

Secondly, we'll need to update the `ObservableQuery.onData` API to support adding callbacks that does not get removed after invocation. This can be indicated through a new optional named parameter, e.g. `removeAfterInvocation` which is defaulted to `true` for backwards compatibility. Then we'll need a way to remove the callbacks we've just created. My suggestion is to make the `onData` method return a `void Function()` which takes care of removing the callbacks added.